### DISCUSSION TITLE
Webend Version inline with Github-release Version

### DISCUSSION THREAD

**Discussion by C9Glax** [dani-garcia/vaultwarden#5274] [General]
Currently the Webend displays the text
```
Vaultwarden Web
Version 2024.6.2 
```
With the recent fixes for the CVEs I got quite the scare, as June is way before the time out the released fixes, just to find that this text is not aligned with any Version/Release on Github for Vaultwarden, and I was in fact running a fixed version. 😌 

What I would like to see is a additional line which displays the Version of Vaultwarden currently running, so something like
```
Vaultwarden Web
Version 2024.6.2
Release 1.32.6
```

**Comment by BlackDex** [comment]
No. This will not happen. 
Some people do not want to show any version at all, so adding an extra will not be that useful. 

You can use the Vaultwarden Backend Admin Interface to view all this.

**Reply by C9Glax** [reply]
I guess that people not wanting to show version at all is a "security" measure?
You are not preventing any malicious intent by hiding something that is only an indicator. If I were to find a vulnerability, I would not stop at a "Oh the version is not shown, guess this instance is secure". I would still attempt to run the exploit, as most likely it will take me a fraction of a second longer than doing a web-scrape if some Version-String on the webpage matches.
Especially on something like Vaultwarden, where the supposed value of the contents FAR outweighs this effort. Even if it takes an hour to run the exploit, a day even, it will still be "economical" to give it a shot.

On the other hand, an indicator that my running version is vulnerable will trigger me to update ASAP to ensure my own peace of mind. I myself already am Auto-Updating daily, and still got scared that it might not have.
Also: The Admin Interface should be inaccessible 99.9% of the time. There is close to zero reason why I should login into it regularly.

**Reply by BlackDex** [reply]
And you seeing the version number everyday will probably also not trigger you or anyone else to check if there are updates or not. So it will not help anyway.

**Reply by C9Glax** [reply]
Then go a step further and add a check if there is a new "latest" tag with version. Make it flashy.

**Reply by kpfleming** [reply]
The admin page already tells you if there are newer versions.
![image](https://github.com/user-attachments/assets/ee2fe68a-0898-4606-a148-0259dbe6eb97)

**Reply by C9Glax** [reply]
Huh, didn't know that.
Point taken.

Still only look at the admin Panel maybe once a year, so might still be worth considering.

**Reply by BlackDex** [reply]
The point is, we will never make everybody happy.
Still the `/api/config` endpoint will at least provide the git hash, and hackers can use that.
There is also the `/api/version` endpoint which everybody can access. So basically if someone really wants, they could still see it relatively easy.

That will still not make you notice there is an update available if we only put the version on the web-vault main page.
With the Dynamic CSS we have implemented it makes it a bit possible now to do this. But not if there is an update available.

I did have some thoughts about this though, ill see if i can work those out a bit more.

**Reply by C9Glax** [reply]
Thank you very much!
